### PR TITLE
Adding new api version 2024-01-01-preview

### PR DESCRIPTION
**Pull request by utkarshjain1508**
# ARM (Control Plane) API Specification Update Pull Request 

> Tip: overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] [ARM review queue] (for **merge** queues, see [2])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

</details>

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the Diagram Step 2, "ARM API changes Review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

File: package-lock.json
Comment (mikeharder) [inline package-lock.json:1]: Revert changes to this file

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (utkarshjain1508) [reply package-lock.json:1]: done

**Comment by utkarshjain1508** [comment]
@rkmanda I have made all the changes on my end. Can you give approval for breaking change ?

**Review by mikeharder — APPROVED** [review]

**Comment by visingla-ms** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Elastic/monitors/{monitorName}/openAIIntegration": {

[ARMBlockingComment]
Please use plural resourceType here, and also since OpenAI is a brand name, changing to lower case is not correct.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:653 in 2c079d9. [](commit_id = 2c079d9dfaa884470b8fdda4d38c97b551219bd9, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "text/json"

ARM APIs only support returning application/json type of response. Other response types are not allowed here.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:663 in 2c079d9. [](commit_id = 2c079d9dfaa884470b8fdda4d38c97b551219bd9, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "application/*+json"

Same here, only application/json is allowed in request payload.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:762 in 2c079d9. [](commit_id = 2c079d9dfaa884470b8fdda4d38c97b551219bd9, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "type": "string",

If you mean AzureOpenAI Resource here, please use [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details).

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:2589 in 2c079d9. [](commit_id = 2c079d9dfaa884470b8fdda4d38c97b551219bd9, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "type": "string"

Please use [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret) here, and make sure not to return the key in GET/PUT/PATCH API response. It can only be returned in Post API Response if needed.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:2599 in 2c079d9. [](commit_id = 2c079d9dfaa884470b8fdda4d38c97b551219bd9, deletion_comment = False)

**Comment by visingla-ms** [comment]
>         "id": {

Why not use ProxyResource definition from here: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json)?

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:2618 in 2c079d9. [](commit_id = 2c079d9dfaa884470b8fdda4d38c97b551219bd9, deletion_comment = False)

**Comment by visingla-ms** [comment]
>       "type": "object",

Please description property here.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:2651 in 2c079d9. [](commit_id = 2c079d9dfaa884470b8fdda4d38c97b551219bd9, deletion_comment = False)

**Comment by visingla-ms** [comment]
>       "type": "object",

Please description property here.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:2659 in 2c079d9. [](commit_id = 2c079d9dfaa884470b8fdda4d38c97b551219bd9, deletion_comment = False)

File: specification/elastic/resource-manager/readme.md [hunk 1]
```diff
@@ -216,6 +225,12 @@ directive:
     where:
       - $.definitions.VMIngestionDetailsResponse.properties.ingestionKey
     reason: Secrets are OK to return in a POST response.
+  - suppress: SECRET_PROPERTY
```
Comment (visingla-ms) [inline specification/elastic/resource-manager/readme.md:228]: Why is this suppression required here?

Comment (utkarshjain1508) [reply specification/elastic/resource-manager/readme.md]: Removed....had added it before due earlier versions.

**Comment by utkarshjain1508** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Elastic/monitors/{monitorName}/openAIIntegration": {
> > ```
> 
> [ARMBlockingComment] Please use plural resourceType here, and also since OpenAI is a brand name, changing to lower case is not correct.
> 
> Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:653 in [2c079d9](https://github.com/Azure/azure-rest-api-specs/commit/2c079d9dfaa884470b8fdda4d38c97b551219bd9). [](commit_id = [2c079d9](https://github.com/Azure/azure-rest-api-specs/commit/2c079d9dfaa884470b8fdda4d38c97b551219bd9), deletion_comment = False)

@visingla-ms need to change it to lowercase because of https://github.com/Azure/azure-openapi-validator/blob/main/docs/path-resource-type-name-camel-case.md

**Comment by utkarshjain1508** [comment]
> > ```
> >     "id": {
> > ```
> 
> Why not use ProxyResource definition from here: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json?
> 
> Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:2618 in [2c079d9](https://github.com/Azure/azure-rest-api-specs/commit/2c079d9dfaa884470b8fdda4d38c97b551219bd9). [](commit_id = [2c079d9](https://github.com/Azure/azure-rest-api-specs/commit/2c079d9dfaa884470b8fdda4d38c97b551219bd9), deletion_comment = False)

@visingla-ms  We just need the resource id of the resource for bookkeeping purposes

**Comment by samsaha-ms** [comment]
>           "201": {

If PUT/POST/PATCH/DELETE responses have location or azure async headers, then MUST have x-ms-long-running-operation set to true.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:791 in 6cd7269. [](commit_id = 6cd7269ebebd6621aee19f607c4ed46542f01f95, deletion_comment = False)

**Comment by samsaha-ms** [comment]
>           "201": {

If it is long running operation then add x-ms-long-running-operation: true 
[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation)

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:791 in 6cd7269. [](commit_id = 6cd7269ebebd6621aee19f607c4ed46542f01f95, deletion_comment = False)

**Comment by samsaha-ms** [comment]
>           "204": {

If it is long running operation then add x-ms-long-running-operation: true 
[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation)

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:840 in 6cd7269. [](commit_id = 6cd7269ebebd6621aee19f607c4ed46542f01f95, deletion_comment = False)

**Comment by samsaha-ms** [comment]
>         "lastRefreshAt": {

Will this be passed in requestBody ever? if it is something only returned by RP in response then marked it as readonly.

---
Refers to: specification/elastic/resource-manager/Microsoft.Elastic/preview/2024-01-01-preview/elastic.json:2591 in 6cd7269. [](commit_id = 6cd7269ebebd6621aee19f607c4ed46542f01f95, deletion_comment = False)

**Comment by samsaha-ms** [comment]
@utkarshjain1508 , Please fix the lint errors present in this PR. https://github.com/Azure/azure-rest-api-specs/pull/27419/checks?check_run_id=21253677293

**Comment by utkarshjain1508** [comment]
> @utkarshjain1508 , Please fix the lint errors present in this PR. https://github.com/Azure/azure-rest-api-specs/pull/27419/checks?check_run_id=21253677293

@samsaha-ms The linting errors are for monitorName property which is not a new property. The error is being flagged in the existing API spec. Adding pattern validation on that property will be a breaking change

**Comment by samsaha-ms** [comment]
> > @utkarshjain1508 , Please fix the lint errors present in this PR. https://github.com/Azure/azure-rest-api-specs/pull/27419/checks?check_run_id=21253677293
> 
> @samsaha-ms The linting errors are for monitorName property which is not a new property. The error is being flagged in the existing API spec. Adding pattern validation on that property will be a breaking change

As discussed, add suppression if you think it would be breaking change for you. after that remove `ArmChangesRequested` for further review.

**Pull request merged**